apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/config/BizConfig.java

```diff
@@ -154,6 +158,21 @@ public int accessKeyAuthTimeDiffTolerance() {
         DEFAULT_ACCESS_KEY_AUTH_TIME_DIFF_TOLERANCE);
   }
 
+  public int releaseHistoryRetentionSize() {
+    int count = getIntProperty("apollo.release-history.retention.size", DEFAULT_RELEASE_HISTORY_RETENTION_SIZE);
+    return checkInt(count, 1, Integer.MAX_VALUE, DEFAULT_RELEASE_HISTORY_RETENTION_SIZE);
```

Comment:
> I think we don't need to use native query here as it may not work well with other databases. I can see the limit offset setup but I suppose it could be replaced with Pageable?

yes, it has been changed to a JPA query now.

> as the default config is -1, I think we could still let the user config -1 but not 0?

The restriction that one record of history must be kept is deliberately designed for this scenario to ensure the basic configuration rollback capability. Perhaps we can explain this point in the documentation, which I will update.